demo/universal/native/lib/Actions.re

```diff
@@ -0,0 +1,296 @@
+/*
+   Due to the issues that you can see at create (line 71), edit (line 136) and delete (line 192) functions we will have the args always as a list of a single list of args.
+   To parse it we need to get this first item that will indeed be the args and pass it further
+ */
+[@platform native]
+let getArgs = body => {
+  Yojson.Basic.from_string(body)
+  |> Yojson.Basic.Util.to_list
+  |> List.hd
+  |> Yojson.Basic.Util.to_list;
+};
+
+module Notes = {
+  [@platform native]
+  let createNoteResponse = note => {
+    React.Json(
+      Note.(
+        `Assoc([
+          ("id", `Int(note.id)),
+          ("title", `String(note.title)),
+          ("content", `String(note.content)),
+          ("updated_at", `Float(note.updated_at)),
+        ])
+      ),
+    );
+  };
+
+  // Lets say this is the server action declared by the end-user
+  // It's going to be on top to this that we are going to generate the codes bellow
+
+  [@platform native]
+  let createHandler = (~title, ~content) => {
+    let note = DB.addNote(~title, ~content);
+    let%lwt response =
+      switch%lwt (note) {
+      | Ok(note) => Lwt.return(createNoteResponse(note))
+      | Error(e) => Lwt.return(React.Json(`String(e)))
+      };
+    Lwt.return(response);
+  };
+  // This the router  handler that will handle parsing the args and calling the handler the user declared
+  // This code will be generated by the ppx automatically
+  [@platform native]
+  let createRouteHandler = body => {
+    // Parse the body to get the args
+    let (title, content) =
+      switch (getArgs(body)) {
+      | [title, content] => (
+          // It would be handle by a title_of_json/content_of_json in the future provided by the end-user and the ppx
+          title |> Yojson.Basic.Util.to_string,
+          content |> Yojson.Basic.Util.to_string,
+        )
+      | _ => failwith(Printf.sprintf("Invalid arguments %s", body))
```

Comment:
Can you deal with errors in all the functions in case of wrong types at runtime?

Reply:
For this, we will need to check for the Error handling on react. We have a task for it on the board.